### ISSUE TITLE
Clean up make install

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-portal#488]
Clean up make install to ensure right files are installed, etc:
 - sudo echo "Starting install of git hash ...", and when done sudo echo "Install done..."
 - Be careful about copying backup, ~, # files to panther; be more careful deleting files. Packaging will get us some of this for free.  'syncp --delete'

 _Imported from trac ticket [#488](http://trac.gpolab.bbn.com/proto-ch/ticket/488),  created by ahelsing on 04-19-2013 at 13:07, last modified: 05-28-2013 at 15:01_

**Comment by tcmitchell** [comment]
`logger` is a command line interface to syslog on Linux. We could theoretically log the start and end of install.

Is it desirable to have the installation appear in the sudo logs? Or in syslog? Or both?


_Trac comment by tmitchel (github user: tcmitchell) on 04-22-2013 at 16:23_

**Comment by ahelsing** [comment]
Tom: it's desirable for infra for you to do `sudo echo`, because then the start/end times show up in the sudo log chronologically sorted for us inline with the other sudo traffic, so we can make reasonable guesses about what we're looking at.

If you also want to do logger, that's certainly fine.

_Trac comment by chaos on 04-22-2013 at 16:25_

**Comment by tcmitchell** [comment]
Adding "sudo echo" will result in the following in the sudo log:
```
Apr 23 14:47:27 dagoola sudo: tmitchel : TTY=pts/0 ; PWD=/home/tmitchel/proto-ch ; USER=root ; COMMAND=/usr/bin/make install
Apr 23 14:47:27 dagoola sudo: tmitchel : TTY=pts/0 ; PWD=/home/tmitchel/proto-ch/example ; USER=root ; COMMAND=/bin/echo install-data-local in\
 example dir
Apr 23 14:47:27 dagoola sudo: tmitchel : TTY=pts/0 ; PWD=/home/tmitchel/proto-ch/example ; USER=root ; COMMAND=/bin/echo install-data-hook in \
example dir
```
No intermediate commands, like directory creation and file installation, are shown.

I'd like to start using GNU autotools to put us on a path to a configurable install, but there's no easy/straightforward way to wrap the install target with a before and after echo. How much effort should we put into this given the utility of the above output?


_Trac comment by tmitchel (github user: tcmitchell) on 04-23-2013 at 14:52_

**Comment by ahelsing** [comment]
Tom and i discussed this: the reason i was keen to have the start/end "echo" statements is because the make install generates a lot of noise in the sudo log, and we need to be able to differentiate between things the make install does (which we can track down by looking in git), and things people do by hand.

However, we think the noise is actually being caused by a number of "sudo" invocations inside subsidiary Makefiles, e.g.
```
sudo /bin/cp <stuff> <place>
```
That shouldn't be necessary --- the `sudo make install` should get all of those subsidiary commands to run with the right privileges.  We think the first thing to try is cleaning up those invocations.  If doing so doesn't introduce any problems and turns off the sudo log noise, then i agree it won't be necessary to insert the "sudo echo" statements.


_Trac comment by chaos on 04-23-2013 at 15:15_

**Comment by tcmitchell** [comment]
This work is complete in development and will be part of a future release.

_Trac comment by tmitchel (github user: tcmitchell) on 05-08-2013 at 13:06_

**Comment by tcmitchell** [comment]
Fixed in 1.5, part of #354.

_Trac comment by tmitchel (github user: tcmitchell) on 05-28-2013 at 14:52_
